### ISSUE TITLE
Amcrest (And possibly Dahua) Doorbells

### ISSUE THREAD

**Issue by itkfilelor** [AlexxIT/go2rtc#247]
Hi there sorry if this is a dupe, I didn't see anything else regarding this. Just want give a heads up that Amcrest doorbells (and likely Dahua since AMcrest is just rebranded Dahua) 'break' with go2rtc unless you turn off the back channel. Once connected, the doorbell assumes you are engaged in a 2-way call and disables the doorbell. I haven't dug in enough to look at toggling the 2-way as I haven't needed it as of yet. so for now I just have it disabled =)

**Comment by AlexxIT** [comment]
Yes. It's dup. And it's known problem of Amcrest doorbells. Button works only with disabled backchannel.

Two way communication should be reworked to optional and delayed enable with reconnection to camera. Needs time.

**Comment by itkfilelor** [comment]
> Yes. It's dup. And it's known problem of Amcrest doorbells. Button works only with disabled backchannel.
> 
> Two way communication should be reworked to optional and delayed enable with reconnection to camera. Needs time.

Oh ok, sorry about that. Wish I saw it before hand (my fault obviously). My db stopped working mysteriously to the point that I bought a new one. When that one also didn't work I banged my head against the wall for about an hour before I figured it out. :grin:  At least now I have a spare to take apart :grinning: 
Might I suggest making it more prominent somewhere in the README until its adjusted? OFC I jumped in on this from Frigate.. so..

**Comment by bagobones** [comment]
I ran into this same bug with my AD110 doorbell, I had no idea what was going on I thought the physical button had stopped working.

**Comment by akmolina28** [comment]
In case anyone needs example code....

```
streams:
  doorbell:
    - rtsp://user:password@X.X.X.X#backchannel=0
    - "ffmpeg:doorbell#audio=opus"
```

This is how I got my Amcrest AD110 set up so that 1-way audio still works in the WebRTC and MSE streams without breaking the doorbell button.

**Comment by brnmckllr** [comment]
> 

I don't know why this works, but it does, can confirm on Amcrest Model AD410. Why the ' #backchannel=0 ' though? That just makes the rest of the line a comment

**Comment by NickM-27** [comment]
> > 
> 
> I don't know why this works, but it does, can confirm on Amcrest Model AD410. Why the ' #backchannel=0 ' though? That just makes the rest of the line a comment

that is just an ide mistake, it is not handled as a comment and go2rtc interprets that to mean it should not open a connection to the backchannel audio that is provided by the stream

**Comment by brnmckllr** [comment]
> > > 
> > 
> > 
> > I don't know why this works, but it does, can confirm on Amcrest Model AD410. Why the ' #backchannel=0 ' though? That just makes the rest of the line a comment
> 
> that is just an ide mistake, it is not handled as a comment and go2rtc interprets that to mean it should not open a connection to the backchannel audio that is provided by the stream

Thank. you! This is my setup now:

    doorbell:
      - rtsp://username:password@X.X.X.X#backchannel=0
      - "ffmpeg:doorbell#audio=opus"
    doorbell_sub:
      - rtsp://username:password@X.X.X.X:554/cam/realmonitor?channel=1&subtype=1#backchannel=0
      - "ffmpeg:doorbell_sub#audio=opus"

I used [Amcrest IP Config Tool](https://support.amcrest.com/hc/en-us/articles/360004527171-Amcrest-IP-Config-Tool-Download) and changed both channels to 264 and I changed the substream to 5 Frame rate and I'm using the substream to detect.

<img width="852" alt="Screenshot 2023-07-06 at 7 19 39 PM" src="https://github.com/AlexxIT/go2rtc/assets/12626850/20694477-5fb5-4e23-857d-3fbec0b15cfc">

I also used this to get ride of the amcrest logo, just because: http://X.X.X.X/cgi-bin/configManager.cgi?action=setConfig&VideoWidget[0].PictureTitle.EncodeBlend=false

**Comment by itkfilelor** [comment]
> > > > 
> > > 
> > > 
> > > I don't know why this works, but it does, can confirm on Amcrest Model AD410. Why the ' #backchannel=0 ' though? That just makes the rest of the line a comment
> > 
> > 
> > that is just an ide mistake, it is not handled as a comment and go2rtc interprets that to mean it should not open a connection to the backchannel audio that is provided by the stream
> 
> Thank. you! This is my setup now:
> 
> ```
> doorbell:
>   - rtsp://username:password@X.X.X.X#backchannel=0
>   - "ffmpeg:doorbell#audio=opus"
> doorbell_sub:
>   - rtsp://username:password@X.X.X.X:554/cam/realmonitor?channel=1&subtype=1#backchannel=0
>   - "ffmpeg:doorbell_sub#audio=opus"
> ```
> 
> I used [Amcrest IP Config Tool](https://support.amcrest.com/hc/en-us/articles/360004527171-Amcrest-IP-Config-Tool-Download) and changed both channels to 264 and I changed the substream to 5 Frame rate and I'm using the substream to detect.
> 
> <img alt="Screenshot 2023-07-06 at 7 19 39 PM" width="852" src="https://user-images.githubusercontent.com/12626850/251611449-20694477-5fb5-4e23-857d-3fbec0b15cfc.png">
> I also used this to get ride of the amcrest logo, just because: [http://X.X.X.X/cgi-bin/configManager.cgi?action=setConfig&VideoWidget[0].PictureTitle.EncodeBlend=false](http://X.X.X.X/cgi-bin/configManager.cgi?action=setConfig&VideoWidget%5B0%5D.PictureTitle.EncodeBlend=false)

If doorbell_sub is ONLY for detect, you can remove the  `- "ffmpeg:doorbell_sub#audio=opus"` line and save yourself some processing... detect takes no audio and you're needlessly converting it.

```    
    front_door: 
      - rtsp://<CREDENTIALS>@doorbell.lan:554/cam/realmonitor?channel=1&subtype=0#backchannel=0
      - ffmpeg:front_door#audio=opus
    front_door_detect:
      - rtsp://<CREDENTIALS>@doorbell.lan:554/cam/realmonitor?channel=1&subtype=1#backchannel=0
```
works perfect for me ... accept when my WiFi is being shyte :grin:
